weaviate/collections/base.py

```diff
@@ -12,17 +13,59 @@
     _collection_configs_from_json,
     _collection_configs_simple_from_json,
 )
-from weaviate.exceptions import UnexpectedStatusCodeException
+from weaviate.exceptions import EmptyResponseException, UnexpectedStatusCodeException
 from weaviate.util import _capitalize_first_letter, _decode_json_response_dict
 
 if TYPE_CHECKING:
     from weaviate.collections.batch.executor import BatchExecutor
 
 
 class _CollectionBase:
-    def __init__(self, name: str) -> None:
+    def __init__(self, connection: Connection, name: str) -> None:
+        self._connection = connection
         self.name = _capitalize_first_letter(name)
 
+    def shards(
+        self,
+    ) -> List[Shard]:
+        """
+        Get the statuses of all the shards of this collection.
+
+        If the collection is multi-tenancy, each shard is one of the tenants. If the collection is single-tenancy, there is only one shard.
```

Comment:
> If the collection is single-tenancy, there is only one shard.

I am pretty sure that this is not true

Reply:
```python
import weaviate
import weaviate.classes as wvc

client = weaviate.connect_to_local()

client.collections.delete_all()

st = client.collections.create(
    name="st",
)
mt = client.collections.create(
    name="mt",
    multi_tenancy_config=wvc.Configure.multi_tenancy()
)

mt.tenants.create([wvc.Tenant(name="t1"), wvc.Tenant(name="t2")])

st.data.insert_many([{} for _ in range(100000)])

print(st.shards())
print(mt.shards())
```
yields:
```bash
[Shard(collection='St', name='1mX0C7bh2rI2', object_count=100000)]
[Shard(collection='Mt', name='t1', object_count=0), Shard(collection='Mt', name='t2', object_count=0)]
```

But the statement in the doc-string is quite strong so I'll remove to avoid any potential for confusion

Reply:
it depends in your config, if you configure a class with more shards it has more shards. If you don't then it is just one :)

Reply:
Oh yeh, the sharding config!